lecture_3/03_project/calculator/internal/infrastructure/sqlite/betRepository.go

```diff
@@ -0,0 +1,186 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	storagemodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/sqlite/models"
+)
+
+// BetRepository provides methods that operate on cals_bets SQLite database.
+type BetRepository struct {
+	dbExecutor DatabaseExecutor
+	betMapper  BetMapper
+}
+
+// NewBetRepository creates and returns a new BetRepository.
+func NewBetRepository(dbExecutor DatabaseExecutor, betMapper BetMapper) *BetRepository {
+	return &BetRepository{
+		dbExecutor: dbExecutor,
+		betMapper:  betMapper,
+	}
+}
+
+// InsertBetCalculated inserts the provided calculated bet into the database. An error is returned if the operation
+// has failed.
+func (r *BetRepository) InsertBetCalculated(ctx context.Context, bet domainmodels.BetCalculated) error {
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryInsertBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "calc_bets repository failed to insert a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryInsertBet(ctx context.Context, bet storagemodels.BetCalculated) error {
+	// If payout is 0, do not insert it.
+	if bet.Payment == 0 {
+		insertBetSQL := "INSERT INTO bets(id, selection_id, selection_coefficient, payment) VALUES (?, ?, ?, ?)"
+		statement, err := r.dbExecutor.PrepareContext(ctx, insertBetSQL)
+		if err != nil {
+			return err
+		}
+
+		_, err = statement.ExecContext(ctx, bet.Id, bet.SelectionId, bet.SelectionCoefficient, bet.Payment)
+		return err
+	}
+
+	insertBetSQL := "INSERT INTO bets(id, selection_id, selection_coefficient, payment) VALUES (?, ?, ?, ?)"
+	statement, err := r.dbExecutor.PrepareContext(ctx, insertBetSQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = statement.ExecContext(ctx, bet.Id, bet.SelectionId, bet.SelectionCoefficient, bet.Payment)
+	return err
+}
+
+// UpdateBetCalculated updates the provided calculated bet in the database. An error is returned if the operation
+// has failed.
+func (r *BetRepository) UpdateBetCalculated(ctx context.Context, bet domainmodels.BetCalculated) error {
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryUpdateBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "calc_bets repository failed to update a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryUpdateBet(ctx context.Context, bet storagemodels.BetCalculated) error {
+	updateBetSQL := "UPDATE bets SET selection_id=?, selection_coefficient=?, payment=? WHERE id=?"
+
+	statement, err := r.dbExecutor.PrepareContext(ctx, updateBetSQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = statement.ExecContext(ctx, bet.Payment, bet.Id)
```

Comment:
Is this going to work? You have only 2 arguments here and 4 arguments few lines above when creating `updateBetSQL`. Let me know if I'm missing something.

Reply:
Thank you, I did not notice, this was a mistake. But the code worked because I actually never called UpdateBet function, now I am going to fix that as well.

Reply:
Yep I know, no worries!